src/charm.py

```diff
@@ -178,6 +186,21 @@ def _on_remove(self, event: RemoveEvent) -> None:
         self._delete_external_upf_service()
 
     def _delete_external_upf_service(self) -> None:
+        # NOTE: We want to preform this removal only if the last remaining unit
```

Comment:
```suggestion
        # NOTE: We want to perform this removal only if the last remaining unit
```

Reply:
Fixed in https://github.com/canonical/sdcore-upf-operator/pull/52/commits/7083a54806325bb35a6545c17705460ff3a0856b